packages/service-library/tests/test_rabbitmq_robust_rpc.py

```diff
@@ -0,0 +1,229 @@
+# pylint:disable=redefined-outer-name
+# pylint:disable=unused-argument
+
+import asyncio
+from typing import Any, Awaitable, Final
+
+import pytest
+from pydantic import NonNegativeInt
+from servicelib.rabbitmq_robust_rpc import (
+    NotStartedError,
+    RemoteMethodNotRegisteredError,
+    RobustRPCClient,
+    RobustRPCServer,
+)
+from settings_library.rabbit import RabbitSettings
+
+pytest_simcore_core_services_selection = [
+    "rabbit",
+]
+
+MULTIPLE_REQUESTS_COUNT: Final[NonNegativeInt] = 100
+
+
+@pytest.fixture
+async def rpc_client(rabbit_service: RabbitSettings) -> RobustRPCClient:
+    client = RobustRPCClient(rabbit_settings=rabbit_service)
+    await client.start()
+    yield client
+    await client.stop()
+
+
+@pytest.fixture
+async def rpc_server(rabbit_service: RabbitSettings) -> RobustRPCServer:
+    server = RobustRPCServer(rabbit_settings=rabbit_service)
+    await server.start()
+    yield server
+    await server.stop()
+
+
+async def add_me(*, x: Any, y: Any) -> Any:
+    result = x + y
+    # NOTE: types are not enforced
+    # result's type will on the caller side will be the one it has here
+    return result
+
+
+class CustomClass:
+    def __init__(self, x: Any, y: Any) -> None:
+        self.x = x
+        self.y = y
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} x={self.x}, y={self.y}>"
+
+    def __eq__(self, other: "CustomClass") -> bool:
+        return self.x == other.x and self.y == other.y
+
+    def __add__(self, other: "CustomClass") -> "CustomClass":
+        return CustomClass(x=self.x + other.x, y=self.y + other.y)
+
+
+@pytest.mark.parametrize(
+    "x,y,expected_result,expected_type",
+    [
+        pytest.param(12, 20, 32, int, id="expect_int"),
+        pytest.param(12, 20.0, 32.0, float, id="expect_float"),
+        pytest.param(b"123b", b"xyz0", b"123bxyz0", bytes, id="expect_bytes"),
+        pytest.param([1, 2], [2, 3], [1, 2, 2, 3], list, id="list_addition"),
+        pytest.param(
+            CustomClass(2, 1),
+            CustomClass(1, 2),
+            CustomClass(3, 3),
+            CustomClass,
+            id="custom_class",
+        ),
+        pytest.param(
+            CustomClass([{"p", "1"}], [{"h": 1}]),
+            CustomClass([{3, b"bytes"}], [{"b": 2}]),
+            CustomClass([{"p", "1"}, {3, b"bytes"}], [{"h": 1}, {"b": 2}]),
+            CustomClass,
+            id="custom_class_complex_objects",
+        ),
+    ],
+)
+async def test_base_rpc_pattern(
+    rpc_client: RobustRPCClient,
+    rpc_server: RobustRPCServer,
+    x: Any,
+    y: Any,
+    expected_result: Any,
+    expected_type: type,
+):
+    await rpc_server.register(add_me)
+
+    request_result = await rpc_client.request(add_me.__name__, x=x, y=y)
```

Comment:
any option to not have ```__name__``` there?

Reply:
this is a test and `__name__` was just convenient. Normally you would call this function from a different process where you might not even have that code.